### PR TITLE
Support max on single-level struct in aggregation context

### LINKED ISSUE #3541: [FEA] Support max on single-level struct in aggregation context
**Is your feature request related to a problem? Please describe.**
I wish the plugin can support Max on Single-struct in aggregation context.

**Additional context**
Code example:
```
import org.apache.spark.sql.Row
import org.apache.spark.sql.types._

val data = Seq(
    Row(Row("Adam ","","Green"),1,"M",10,"Math:13,Gym:24,English:45",Map("hair"->"black","eye"->"black"),"{\"hair\":\"black\",\"eye\":\"black\"}"),
    Row(Row("Bob ","Middle","Green"),2,"M",20,"Math:55,Gym:24,English:37",Map("hair"->"yellow","eye"->"yellow"),"{\"hair\":\"green\",\"eye\":\"green\"}"),
    Row(Row("Cathy ","","Green"),3,"F",30,"Math:83,Gym:15,English:63",Map("hair"->"blue","eye"->"blue"),"{\"hair\":\"blue\",\"eye\":\"blue\"}")
)

val schema = (new StructType()
  .add("name",new StructType()
    .add("firstname",StringType)
    .add("middlename",StringType)
    .add("lastname",StringType)) 
  .add("low",IntegerType)
  .add("gender",StringType)
  .add("high",IntegerType)
  .add("score",StringType)
  .add("feature",MapType(StringType,StringType))
  .add("feature_json",StringType))

val df = spark.createDataFrame(spark.sparkContext.parallelize(data),schema)
df.write.format("parquet").mode("overwrite").save("/tmp/yl/tmpdatas")

val df2 = spark.read.parquet("/tmp/yl/tmpdatas")
df2.createOrReplaceTempView("df")


spark.sql("SELECT max(name) from df").show()
```

### PR DESCRIPTION
**Pull request by res-life**
[FEA] Support max on single-level struct in aggregation context #3541

This fixes #3541 

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

File: integration_tests/src/main/python/hash_aggregate_test.py [hunk 1]
```diff
@@ -1695,3 +1695,22 @@ def test_groupby_std_variance_partial_replace_fallback(data_gen,
         exist_classes=','.join(exist_clz),
         non_exist_classes=','.join(non_exist_clz),
         conf=local_conf)
+
+@ignore_order
+@pytest.mark.parametrize('data_type',  all_gen + [NullGen()], ids=idfn)
```
Comment (firestarman) [inline integration_tests/src/main/python/hash_aggregate_test.py:1700]:
```suggestion
@pytest.mark.parametrize('data_gen',  all_gen + [NullGen()], ids=idfn)
```

**Review by firestarman — COMMENTED** [review]
Overall it looks good to me, but a NIT.

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

File: docs/supported_ops.md [hunk 2]
```diff
@@ -15227,7 +15227,7 @@ are limited.
 <td><b>NS</b></td>
 <td><b>NS</b></td>
 <td> </td>
-<td><b>NS</b></td>
+<td><em>PS<br/>UTC is only supported TZ for child TIMESTAMP;<br/>unsupported child types BINARY, CALENDAR, ARRAY, STRUCT, UDT</em></td>
```
Comment (revans2) [inline docs/supported_ops.md:15230]: This says that it works for window operations too, but I see no tests for window operations.

Comment (res-life) [reply docs/supported_ops.md]: Done, removed the window support.

File: integration_tests/src/main/python/hash_aggregate_test.py [hunk 3]
```diff
@@ -1695,3 +1695,22 @@ def test_groupby_std_variance_partial_replace_fallback(data_gen,
...
         conf=local_conf)
+
+@ignore_order
+@pytest.mark.parametrize('data_gen',  all_gen + [NullGen()], ids=idfn)
+def test_max_single_level_struct(data_gen):
+    df_gen = [
+        ('a', StructGen([
+                ('aa', data_gen),
+                ('ab', data_gen)])),
+        ('b', IntegerGen())]
```
[the remark this answers is not in the document]
Comment (res-life) [reply integration_tests/src/main/python/hash_aggregate_test.py]: Done,  updated.

[the remark this answers is not in the document]
Comment (res-life) [reply integration_tests/src/main/python/hash_aggregate_test.py]: Suported both Min and Max.
Removed the window support.
Updated test case.
Seems CUDF has a bug: https://github.com/rapidsai/cudf/issues/8974#issuecomment-1006400752. After confired, I'll file a issue.

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
Suggest wait, @ttnghia is working on the fix of https://github.com/rapidsai/cudf/issues/8974#issuecomment-1006400752

**Comment by res-life** [comment]
@firestarman  @revans2  help to review, is not blocked now.

**Review by revans2 — APPROVED** [review]

**Pull request merged**